### PR TITLE
Basic Offense Play

### PR DESCRIPTION
**Pull request by kfu02**
## Description

Fills in basic122 with a sensible default offense play: move two robots to the wings, have a third robot pass to one of those two, and shoot once the wings have received a pass.

Costs are now passed into the pass tactic, to allow different plays to choose how they want to pass differently.

Currently the passing is buggy because kick speed is wrong and capture planner is weak, need to fix in separate PRs.

## Steps to test
Run sim.

Expected result: Description of sensible default offense play matches observed behavior.

### CODE REVIEW

File: rj_gameplay/rj_gameplay/play/basic122.py [hunk 1]
```diff
@@ -1,116 +1,138 @@
...
+
+            return self.get_robot_intents(world_state)
+
+        elif self._state == State.INIT_PASS:
+            init_passer_cost = stp.role.cost.PickClosestToPoint(world_state.ball.pos)
+            init_receiver_cost = stp.role.cost.PickClosestToPoint(self._seek_pts[0])
+            self.prioritized_tactics = [
+                goalie_tactic.GoalieTactic(world_state, 0),
+                pass_tactic.PassTactic(
+                    world_state, init_passer_cost, init_receiver_cost
```
Comment (Yudai-8-om) [inline rj_gameplay/rj_gameplay/play/basic122.py:85]: Is there a reason that we don't have seek tactic here?

File: rj_gameplay/rj_gameplay/tactic/striker_tactic.py [hunk 2]
```diff
@@ -8,213 +8,59 @@
...
+            self.init_roles(world_state)
 
-        if capture_result and capture_result[0].is_filled():
-            return [self.capture]
-        if shoot_result and shoot_result[0].is_filled():
-            self.shoot.skill.target_point = find_target_point(
-                world_state, kick_speed=KICK_SPEED
-            )
-            return [self.shoot]
+        return [(role.robot.id, role.tick(world_state)) for role in self.assigned_roles]
```
Comment (Yudai-8-om) [inline rj_gameplay/rj_gameplay/tactic/striker_tactic.py:43]: low priority, but we don't need for loop here

**Review by Yudai-8-om — APPROVED** [review]
Works great overall.
Just wanna note that the receiver starts capturing too early, so we might wanna fix that later if kick speed didn't help

File: rj_gameplay/rj_gameplay/play/basic122.py (hunk 1 above)
Comment (kfu02) [reply rj_gameplay/rj_gameplay/play/basic122.py:85]: no, will add TODO

File: rj_gameplay/stp/play/__init__.py [hunk 3]
```diff
@@ -38,6 +38,10 @@ class IPlay(ABC):
     pass
 
 
+class RoleAssignFailure(Exception):
+    pass
```
Comment (p-nayak11) [inline rj_gameplay/stp/play/__init__.py:42]: Why are were we passing the role assign failure?

**Review by p-nayak11 — APPROVED** [review]
Sim ran fine yesterday, but it's currently giving me trouble. It is probably a local problem. I don't understand why we are passing the role assign failure. Other than that, it looks good.

**Pull request merged**